ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -472,4 +476,25 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
             });
         }
     }
+
+    private saveLastVisitedSectionIndex(sectionIndex: number): void {
+      if (this.shouldSaveLastStep && sectionIndex > this.model.sectionIndex) {
+        this.serviceAgent.saveLastVisitedActivitySection(this.studyGuid, this.activityGuid, this.currentSectionIndex);
+      }
+    }
+
+    private getFirstSectionIndex(): void {
+      this.getIsLoaded$()
+        .pipe(
+          filter(value => !!value),
+          take(1),
+          tap(() => this.shouldSaveLastStep = this.model.statusCode !== this.COMPLETE &&
+            this.config.usesVerticalStepper.includes(this.model.activityCode))
+        )
+        .subscribe(() => {
+          this.currentSectionIndex = this.shouldSaveLastStep
+            ? this.model.sectionIndex || 0
+            : 0;
+        });
+    }
```

Comment:
A few suggestions here:
1. The prefix "get" in a method generally means that it is a property "getter". This is not the intent of this method and it's return type is `void`. I would suggest renaming to something like `initStepperState` or something to that effect.
2. The reference to this.COMPLETE can hopefully be replace with the enum mentioned earlier.
3. Looks like this method intent is to set 2 instance variables, but one is being set in a `tap` and the other in the `subscribe`. Might be clearer if `this.currentSectionIndex` is set in another `tap` and leave the `subscribe` without an argument.
4. Does not look to me that checking the value of `this.shouldSaveLastStep` to decide between and 0 and `model.sectionIndex` helps. We are either going to look at it or not? Consider simplifying to something like:
    `this.currentSectionIndex = this.model.sectionIndex === null ? 0 :  this.model.sectionIndex`

(I don't know if elvis operator is available now in TypeScript. that would be cleaner if can use it`.